### PR TITLE
Feature/active crop ids

### PR DESCRIPTION
**Pull request by Adamws33**
updated active crop data in redux to be an array of ids.

### CODE REVIEW

File: src/pages/CropSidebar/CropSidebar.js [hunk 1]
```diff
@@ -176,7 +176,8 @@ const CropSidebar = ({
 
       return true;
     });
-    dispatchRedux(updateActiveCropData(filtered));
+    console.log('HERE 1', filtered.map((filter) => filter.id));
```
Comment (mikahpinegar) [inline src/pages/CropSidebar/CropSidebar.js:179]: remove

File: src/reduxStore/cropSlice.js [hunk 2]
```diff
@@ -48,6 +48,7 @@ export const updateDateRange = ({ startDate, endDate }) => ({
 const cropDataReducer = (state = initialState, action = null) => {
   switch (action.type) {
     case 'UPDATE_ACTIVE_CROP_DATA':
+      console.log('UPDATE_ACTIVE_CROP_DATA', action.payload.value);
```
Comment (mikahpinegar) [inline src/reduxStore/cropSlice.js:51]: remove

File: src/shared/constants.js [hunk 3]
```diff
@@ -478,22 +478,30 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
         .slice()
         .reverse()
         .forEach((g) => {
+          console.log('crop', b.data.Goals[g].values[0], a.data.Goals[g].values[0]);
```
Comment (mikahpinegar) [inline src/shared/constants.js:481]: remove

File: src/shared/constants.js [hunk 4]
```diff
@@ -478,22 +478,30 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
...
           if (b.data.Goals[g]) {
             bAvg = +b.data.Goals[g].values[0] + bAvg;
           }
           if (a.data.Goals[g]) {
             aAvg = +a.data.Goals[g].values[0] + aAvg;
           }
         });
+      console.log('gggggggggg', aAvg, bAvg);
```
Comment (mikahpinegar) [inline src/shared/constants.js:489]: remove

File: src/shared/constants.js [hunk 5]
```diff
@@ -478,22 +478,30 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
...
 
       aAvg /= selectedItems.length;
       bAvg /= selectedItems.length;
 
+      console.log('averages', aAvg, bAvg);
```
Comment (mikahpinegar) [inline src/shared/constants.js:494]: remove

File: src/shared/constants.js [hunk 6]
```diff
@@ -478,22 +478,30 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
...
+
       if (aAvg > bAvg) {
+        console.log('-1');
```
Comment (mikahpinegar) [inline src/shared/constants.js:497]: remove

File: src/shared/constants.js [hunk 7]
```diff
@@ -478,22 +478,30 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
...
         return -1;
       } if (aAvg === bAvg) {
+        console.log('0');
```
Comment (mikahpinegar) [inline src/shared/constants.js:500]: remove

File: src/shared/constants.js [hunk 8]
```diff
@@ -478,22 +478,30 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
...
         return 0;
       }
+      console.log('1');
```
Comment (mikahpinegar) [inline src/shared/constants.js:503]: remove

**Comment by mikahpinegar** [comment]
unselecting a filter in the sidebar isn't removing the filter

**Comment by mikahpinegar** [comment]
i think that only applies in environmental tolerances for some reason though

**Comment by YazidZidane** [comment]
I got this error at Selector, when I'm trying to sort crops by Average Goal Rating
```
TypeError: Cannot read properties of undefined (reading 'values')
    at http://localhost:3000/static/js/bundle.js:23382:72
    at Array.forEach (<anonymous>)
    at http://localhost:3000/static/js/bundle.js:23381:39
    at Array.sort (<anonymous>)
    at sortCrops (http://localhost:3000/static/js/bundle.js:23378:11)
    at sortByAverageGoals (http://localhost:3000/static/js/bundle.js:5706:65)
    at selectSortingAlgo (http://localhost:3000/static/js/bundle.js:5728:7)
    at handleChange (http://localhost:3000/static/js/bundle.js:49584:7)
    at http://localhost:3000/static/js/bundle.js:57009:9
    at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:135599:18)
```
It's from constants.js line 481.

**Comment by Adamws33** [comment]
> I got this error at Selector, when I'm trying to sort crops by Average Goal Rating
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'values')
>     at http://localhost:3000/static/js/bundle.js:23382:72
>     at Array.forEach (<anonymous>)
>     at http://localhost:3000/static/js/bundle.js:23381:39
>     at Array.sort (<anonymous>)
>     at sortCrops (http://localhost:3000/static/js/bundle.js:23378:11)
>     at sortByAverageGoals (http://localhost:3000/static/js/bundle.js:5706:65)
>     at selectSortingAlgo (http://localhost:3000/static/js/bundle.js:5728:7)
>     at handleChange (http://localhost:3000/static/js/bundle.js:49584:7)
>     at http://localhost:3000/static/js/bundle.js:57009:9
>     at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:135599:18)
> ```
> 
> It's from constants.js line 481.

I was not able to replicate this error. can you tell me the goals you selected, along with the region and the zone so i can try to locate the inconsistency

**Comment by YazidZidane** [comment]
> > I got this error at Selector, when I'm trying to sort crops by Average Goal Rating
> > ```
> > TypeError: Cannot read properties of undefined (reading 'values')
> >     at http://localhost:3000/static/js/bundle.js:23382:72
> >     at Array.forEach (<anonymous>)
> >     at http://localhost:3000/static/js/bundle.js:23381:39
> >     at Array.sort (<anonymous>)
> >     at sortCrops (http://localhost:3000/static/js/bundle.js:23378:11)
> >     at sortByAverageGoals (http://localhost:3000/static/js/bundle.js:5706:65)
> >     at selectSortingAlgo (http://localhost:3000/static/js/bundle.js:5728:7)
> >     at handleChange (http://localhost:3000/static/js/bundle.js:49584:7)
> >     at http://localhost:3000/static/js/bundle.js:57009:9
> >     at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:135599:18)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It's from constants.js line 481.
> 
> I was not able to replicate this error. can you tell me the goals you selected, along with the region and the zone so i can try to locate the inconsistency

I just have another try. My selected Location is Raleigh downtown(default value for the map), and all the other value is also default, there are several goals that would cause this issue like "Forage Harvest Value", "Growing Window", "Biological Nitrogen Fixation", and several other goals.

File: src/pages/CropSidebar/Filters/Filters.js [hunk 9]
```diff
@@ -19,19 +19,19 @@ const DollarsAndRatings = ({ filter, handleChange }) => {
 
   return (
     <div style={style}>
-      {new Array(filter.values.length)
+      {new Array(5)
```
Comment (mikahpinegar) [inline src/pages/CropSidebar/Filters/Filters.js:22]: they aren't all 5 though right? this will break the currency ratings i think

Comment (mikahpinegar) [reply src/pages/CropSidebar/Filters/Filters.js:22]: it's not showing the dollar signs anymore but it didn't do what i expected and have 5 buttons to click

**Review by mikahpinegar — APPROVED** [review]

**Pull request merged**